libass/ass.c

```diff
@@ -1520,19 +1520,43 @@ ASS_Track *ass_new_track(ASS_Library *library)
     return NULL;
 }
 
-int ass_track_set_feature(ASS_Track *track, ASS_Feature feature, int enable)
-{
-    switch (feature) {
-    case ASS_FEATURE_INCOMPATIBLE_EXTENSIONS:
-        //-fallthrough
+static const uint64_t supported_features =
 #ifdef USE_FRIBIDI_EX_API
-    case ASS_FEATURE_BIDI_BRACKETS:
-        track->parser_priv->bidi_brackets = !!enable;
+    1LL << ASS_FEATURE_BIDI_BRACKETS |
 #endif
-        return 0;
-    default:
+    0;
+
+int ass_track_set_feature(ASS_Track *track, ASS_Feature feature, int enable)
+{
+    uint64_t requested;
+
+    if (feature >= 64)
         return -1;
-    }
+
+    requested = 1LL << feature;
+
+    if (feature == ASS_FEATURE_INCOMPATIBLE_EXTENSIONS)
+        requested = supported_features;
+    else if (!(requested & supported_features))
+        return -1;
+
+    if (enable)
+        track->parser_priv->features |= requested;
+    else
+        track->parser_priv->features &= ~requested;
+
+    return 0;
+}
+
+int ass_track_get_feature_state(ASS_Track *track, ASS_Feature feature)
```

Comment:
I think it's better to name it `ass_track_get_feature` without `state` for symmetry.